In [1]:
# Import libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score, confusion_matrix, ConfusionMatrixDisplay

# Intro to Machine Learning ideas
This exercise will introduce you to some common ML ideas such as: 
* supervised learning 
* features and labels
* classifier
* training set, testing set, predictions validation/evaluation 
* performance and confusion matrix

We will use a dataset called "irisdata" for this exercise.

This data contains information about the shape of different samples of iris flowers such as: sepal length, sepal width, petal length, petal width; and finally the species that the sample belongs to.

## Labels and features

In machine learning, we call the species the **label** or **class**, and the other columns **features** or **attributes**.

Now, let's open the data and take a look at what are they.

In [3]:
# Load dataset into a dataframe
df = pd.read_csv("irisdata.csv")

# Display the first few rows of the dataset
print(df.head())

   sepal_length  sepal_width  petal_length  petal_width species
0           5.1          3.5           1.4          0.2  setosa
1           4.9          3.0           1.4          0.2  setosa
2           4.7          3.2           1.3          0.2  setosa
3           4.6          3.1           1.5          0.2  setosa
4           5.0          3.6           1.4          0.2  setosa


#### Goal: The model look at the attributes of one iris and be able to answer correctly which species is the iris. 

For example, we look at row number 3 above. We expect if we ask them: "This iris' sepal length is 4.6, sepal width is 3.1, (other attributes and their values), what is its species?", they can answer: "It's a setosa iris!".

Notice that we are giving them the **attributes values** in our question.

The model's answer or output is a **prediction**. 

The problem that requires our model to give predictions as a category is called a **classification problem**, and our model is interchangably called a **classifier**.

## Training 

Now, to reach our goal, we first have to let our model **learn** the data. 

The main idea is that we will let the model sees the data and the correct answers as well. Depending on the algorithm used, the model will learn on this data. The data we give them to learn will be the **training set**.
